src/plugins/irc/irc-server.c

```diff
@@ -4268,38 +4270,6 @@ irc_server_fingerprint_str_sizes ()
     return strdup (str_sizes);
 }
 
-/*
- * Compares two fingerprints: one hexadecimal (given by user), the second binary
- * (received from IRC server).
- *
- * Returns:
- *    0: fingerprints are the same
- *   -1: fingerprints are different
- */
-
-int
-irc_server_compare_fingerprints (const char *fingerprint,
-                                 const unsigned char *fingerprint_server,
-                                 ssize_t fingerprint_size)
-{
-    ssize_t i;
-    unsigned int value;
-
-    if ((ssize_t)strlen (fingerprint) != fingerprint_size * 2)
-        return -1;
-
-    for (i = 0; i < fingerprint_size; i++)
-    {
-        if (sscanf (&fingerprint[i * 2], "%02x", &value) != 1)
-            return -1;
-        if (value != fingerprint_server[i])
-            return -1;
-    }
-
-    /* fingerprints are the same */
-    return 0;
-}
-
```

Comment:
By removing this function you have done two things:

1. The `sscanf` with `%02x` allows fingerprint option values to use both lower- and uppercase. By using `strcmp` that is no longer the case and case starts to matter.

2. GnuTLS provides all fingerprints/hashes as raw arrays. This function intentionally compared those with a textual representation. By using `strcmp` you were forced to use `sprintf` below to turn all raw arrays into strings just for comparison. That's a slightly odd way of comparing raw byte arrays.

Reply:
I can use `strcasecmp` to make it case-insensitive again if you like. I stand by making it easier for libc to compare them; plus, it's fewer lines, and fewer ways to break it in future.

Reply:
It absolutely needs to be backwards compatible. We don't want to break existing configuration for some users for no reason.

Reply:
Resolved in cb355e4eb76ef4bc1da5a821e353b4e7c10251c5